### PR TITLE
histogram

### PR DESCRIPTION
**Pull request by joaozenobio**

### CODE REVIEW

**Comment by joaozenobio** [comment]
Sorry, I've changed of computer and pre-commit was not installed. After the review I'll fix it.

**Comment by joaozenobio** [comment]
- [ ] #7717

**Comment by ZoeCD** [comment]
Hi! Thank you for your contribution! It seems that only some merge conflicts and lint errors need to be resolved in order to merge this PR. Otherwise, it looks good! 🙂

**Comment by joaozenobio** [comment]
Hi @ZoeCD, I have to wait for your review to fix the issues or should I make the changes and post it here? If the latter, how can I modify the pull request?

**Comment by joaozenobio** [comment]
Forgot to thank you for the response! It's good to hear my contribution looks good.

**Comment by ZoeCD** [comment]
When you push new commits to your fork, the PR is updated automatically. Merry Christmas to you too and happy new year! 😊

**Comment by joaozenobio** [comment]
Hi @ZoeCD! I've made some changes on the PR. I think all the lint issues are gone and there are no merge conflicts now. If there still any changes I need to do, please let me know and I'll try to fix them right away. Thanks for the support and happy new year! :fireworks:

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py [hunk 1]
```diff
@@ -74,6 +74,84 @@ def statistical_dtype_values(draw, *, function):
...
+
+
+@handle_test(
+    fn_tree="functional.ivy.experimental.histogram",
+    private_data_generated=_statistical_dtype_xs_bins_range_axis_castable(),
+    extend_lower_interval=st.booleans(),
+    extend_upper_interval=st.booleans(),
+    density=st.booleans(),
+)
+def test_histogram(
```
Comment (ZoeCD) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py:133]: Hi! Hope you're doing well! I'm adding this comment to "add a review" so it's easier for us to know when you finished the changes I asked you for 😊

**Review by ZoeCD — CHANGES_REQUESTED** [review]

**Comment by joaozenobio** [comment]
Thanks! I'm trying to fix the issue, but I had some family problems last week. I think I'll be able to continue working on it this weekend.

**Comment by ZoeCD** [comment]
I'm sorry to hear that. Hope you and your family are doing well now. Ping me once you're able to continue 😊

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py (hunk 1 above)
Comment (joaozenobio) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py:113]: Hi @ZoeCD! I could solve the problem and other errors that came after, but I got stuck now, because the test function is consuming too much memory and no errors are detected. I've tried looking for help in discord, but I still couldn't figure out what's the matter with my solution. Could you you help, please? I've pushed the last modification to my fork.

**Comment by joaozenobio** [comment]
Hi @ZoeCD ! Just mentioning you here again, because I've commented on the review and I don't know if you've been notified. Hugs! 😁

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py (hunk 1 above)
Comment (ZoeCD) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py:113]: Hi! Could you please share what message you are getting? Is it specifically from this test function or is it happening in other tests too?

Comment (joaozenobio) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py:113]: Hi @ZoeCD! The test_histogram function runs into this error:

```
/usr/bin/python3 /opt/.pycharm_helpers/pycharm/_jb_pytest_runner.py --target test_statistical.py::test_histogram 
Testing started at 14:51 ...
Launching pytest with arguments test_statistical.py::test_histogram --no-header --no-summary -q in /ivy/ivy_tests/test_ivy/test_functional/test_experimental/test_core

Use Database in ReadOnly Mode with local caching !
============================= test session starts ==============================
collecting ... collected 4 items

test_statistical.py::test_histogram[cpu-ivy.functional.backends.numpy-False-False] 
Process finished with exit code 137 (interrupted by signal 9: SIGKILL)
```

When I checked the memory usage, it was consuming all 32GB I have on my computer. 

I can't confirm this error isn't happening with other experimental test functions from the statistical folder because they are getting errors that need to be corrected before running the test, like some fixture errors.

Comment (joaozenobio) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py:117]: @ZoeCD I've pushed the last version I tried to run.

**Comment by danielmunioz** [comment]
HI! @joaozenobio, The memory error you're experiencing is most likely due to the size of the values in your arrays, I'd recommend you tweaking the `min_value` and `max_value` parameters when you're drawing values from the `helpers.dtype_values_axis` function (for example, I tried running the test using min_value=1 and max_value=10 and managed to get It working locally, but these may not the most optimal values for running an effective test)

**Comment by joaozenobio** [comment]
Hi @danielmunioz! Thanks for your response. I've followed your instructions and could solve the problem. Other errors appeared and I'm already dealing with them. As soon as I deal with the test erros I'll exclude my modifications in the .idea folder. Thanks again!

**Comment by joaozenobio** [comment]
Hi @ZoeCD @danielmunioz! I've done the modifications and now I have some doubts on how to continue the developing the function. I've noticed that, when using axis to collapse specific dimensions, Numpy keep dimensions on the output while Tensorflow doesn't. I'm using the default axis argument for tensorflow and numpy.apply_along_axis() for Numpy to perform the function over the axis.

For example:
In this case, the size of the new dimension to be created is 4.
Tensorflow -> input.shape = (2, 1, 2) / axis = 2 / output.shape = (4, 2, 1)
Numpy -> input.shape = (2, 1, 2) / axis = 2 / output.shape = (2, 1, 4)

I couldn't find other way on how to gather data from a specific axis in tensors that are already optimized for the backend and the only manually coded implementation I could find on how to implement the axis argument commes from numpy.apply_along_axis() documentation.

The good news is that Tensorflow is now passing on the tests.

Another thing is that I this pull request doesn't show any changes on the .idea folder. Could you please specify where could you find I was modifying it? This way I think I'll be able to fix the problem.

Sorry for the long message, thank you both, I'll be waiting fot the answer.

**Comment by joaozenobio** [comment]
Hi @danielmunioz! I could solve the axis problem with numpy. Both tensorflow and numpy has passed the tests. However, inside jax version of numpy the function ndindex, that I used to solve the axis problem, does not exist. Should I implement the behaviour of this function manually or raise an issue in jax library?

**Comment by danielmunioz** [comment]
Hi! @joaozenobio, looking at the function it seems that all it does is just iterating over the indexes right? if so yeah, I think this behavior should be able to be replicated quite easily either using a for loop or a custom iterator or maybe, and there is probably an easy workaround as well 
About raising an issue in Jax, I'm not sure if they're planning on supporting this kind of functions, I haven't been able to find its implementation in the source code so probably numpy is importing this ndindex function from somewhere else, maybe it's a c++ implementation or a third party one, but yeah in any case, you can totally try to open an issue over there and discuss the implementation of jax.numpy.ndindex with them!

**Comment by joaozenobio** [comment]
Hi @danielmunioz! I was able to solve the axis problem. I tried to solve it like tf, but I don't know if it is as efficient. Now only torch is still not passing the test, because I'm having problems with its "bins" argument. I'll try to fix it and if I get stuck I'll come back here. :)

**Comment by joaozenobio** [comment]
I accidently asked for another review from @ZoeCD , sorry fot he mistake.
